drivers/can/can_mcan.h

```diff
@@ -700,6 +1278,116 @@ static inline int can_mcan_sys_write_reg(mm_reg_t base, uint16_t reg, uint32_t v
 	return 0;
 }
 
+/**
+ * @brief Bosch M_CAN driver front-end callback helper for reading from memory mapped Message RAM
+ *
+ * @param base Base address of the Message RAM for the given Bosch M_CAN instance. The base address
+ *        must be 32-bit aligned.
+ * @param offset Offset from the start of the Message RAM for the given Bosch M_CAN instance. The
+ *        offset must be 32-bit aligned.
+ * @param[out] dst Destination for the data read. The destination address must be 32-bit aligned.
+ * @param len Number of bytes to read. Must be a multiple of 4.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO General input/output error.
+ */
+static inline int can_mcan_sys_read_mram(mem_addr_t base, uint16_t offset, void *dst, size_t len)
+{
+	volatile uint32_t *src32 = (volatile uint32_t *)(base + offset);
+	uint32_t *dst32 = (uint32_t *)dst;
+	size_t len32 = len / sizeof(uint32_t);
+
+	__ASSERT(base % 4U == 0U, "base must be a multiple of 4");
+	__ASSERT(offset % 4U == 0U, "offset must be a multiple of 4");
+	__ASSERT(POINTER_TO_UINT(dst) % 4U == 0U, "dst must be 32-bit aligned");
+	__ASSERT(len % 4U == 0U, "len must be a multiple of 4");
+
+#if defined(CONFIG_CACHE_MANAGEMENT) && defined(CONFIG_DCACHE)
+	int err;
+
+	err = sys_cache_data_invd_range((void *)(base + offset), len);
+	if (err != 0) {
+		return err;
+	}
+#endif /* !defined(CONFIG_CACHE_MANAGEMENT) && defined(CONFIG_DCACHE) */
+
+	while (len32-- > 0) {
+		*dst32++ = *src32++;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Bosch M_CAN driver front-end callback helper for writing to memory mapped Message RAM
+ *
+ * @param base Base address of the Message RAM for the given Bosch M_CAN instance. The base address
+ *        must be 32-bit aligned.
+ * @param offset Offset from the start of the Message RAM for the given Bosch M_CAN instance. The
+ *        offset must be 32-bit aligned.
+ * @param src Source for the data to be written. The source address must be 32-bit aligned.
+ * @param len Number of bytes to write. Must be a multiple of 4.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO General input/output error.
+ */
+static inline int can_mcan_sys_write_mram(mem_addr_t base, uint16_t offset, const void *src,
```

Comment:
Just out of curiosity, why 'static' ?
Redefinition is prevented by 'inline'. For me 'static' is the indicator that it is only used within this file. Sure, the preprocessor include let the function appear in the file, but it somehow feels wrong.
Am I missing some other quirk from static?

Reply:
Please see https://stackoverflow.com/questions/21835664/why-declare-a-c-function-as-static-inline

Reply:
Ahhh, we want to have the possibility that the compiler can throw away the global visible function and inline the ones used.

Reply:
Exactly.